packages/govern-lib/src/GovernLib.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  Address,
+  DaoData,
+  Networkish,
+  RegistryEntryData,
+  GovernQueueData,
+} from './types'
+import { ErrorInvalidNetwork, ErrorUnexpectedResult } from './errors'
+import NetworkContext from './utils/NetworkContext'
+import GraphqlClient from './GraphqlClient'
+import {
+  QUERY_DAO,
+  QUERY_DAOS,
+  QUERY_REGISTRY_ENTRY,
+  QUERY_REGISTRY_ENTRIES,
+  QUERY_QUEUE,
+  QUERY_QUEUES,
+  QUERY_QUEUES_BY_DAO,
+} from './thegraph-queries'
+
+export type GovernLibConfig = {
+  network: Networkish
+  subgraphUrl?: string
+  verbose?: boolean
+}
+
+class GovernLib {
+  #gql: GraphqlClient
+  readonly config: GovernLibConfig
+  readonly networkContext: NetworkContext
+
+  constructor(config: GovernLibConfig) {
+    this.config = config
+    this.networkContext = NetworkContext.from(
+      config.network,
+      config.subgraphUrl
+    )
+
+    if (!this.networkContext.subgraphUrl) {
+      throw new ErrorInvalidNetwork(
+        `A subgraphUrl couldn’t be found for the chain ID ${this.networkContext.chainId}. ` +
+          `Please specify a subgraphUrl or use a supported chainId.`
+      )
+    }
+
+    this.#gql = new GraphqlClient(this.networkContext.subgraphUrl, {
+      verbose: config.verbose,
+    })
+  }
+
+  private async fetchResult<R>(
+    queryAndParams:
+      | [any]
+      | [
+          any, // TODO: use the type returned by gql``
```

Comment:
``DocumentNode``

Reply:
I remember having an issue with `DocumentNode` at the time this TODO was written, will try again :+1: